api/features/models.py

```diff
@@ -370,14 +366,34 @@ def previous_feature_state_value(self):
         except ObjectDoesNotExist:
             return None
 
+    @property
+    def type(self) -> str:
+        if self.identity_id and self.feature_segment_id is None:
+            return IDENTITY
+        elif self.feature_segment_id and self.identity_id is None:
+            return FEATURE_SEGMENT
+        elif self.identity_id and self.feature_segment_id is None:
+            return ENVIRONMENT
+
+        logger.error(
+            "FeatureState %d does not have a valid type. Defaulting to environment.",
+            self.id,
+        )
+        return ENVIRONMENT
+
     @hook(BEFORE_CREATE)
-    def check_for_existing_env_feature_state(self):
+    def check_for_existing_feature_state(self):
```

Comment:
Maybe worth adding a test for this?

Reply:
I've added more tests to the `FeatureStateTest` class to verify that exceptions are raised on save.